### PR TITLE
Auto-association modals for contents

### PR DESCRIPTION
**Pull request by Wang-Yu-Chao**
Still testing. Just for presentation.

### Summary of Changes

None

### Testing Instructions

None

### Expected result

None

### Actual result

image
image

### Documentation Changes Required

None

### CODE REVIEW

File: administrator/components/com_associations/View/Auto/HtmlView.php [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+				if (empty($support['level']))
+				{
+					unset($this->activeFilters['level']);
+					$this->filterForm->removeField('level', 'filter');
+				}
+				if (empty($support['acl']))
+				{
+					unset($this->activeFilters['access']);
+					$this->filterForm->removeField('access', 'filter');
+				}
```
Comment (jreys) [inline administrator/components/com_associations/View/Auto/HtmlView.php]: So much repeated code, couldn't you make a function for it?

File: administrator/components/com_associations/Model/AutoModel.php [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+			$query->select($db->quoteName('mt.title', 'menutype_title'))
+				->join(
+					'LEFT', $db->quoteName('#__menu_types', 'mt') . ' ON ' . $db->quoteName('mt.menutype')
+					. ' = ' . $db->quoteName('c2.' . $menutypeField)
+				);
+		}
+
+		$query->where('(c.' . $pk . ' = ' . (int) $referenceId . ' OR c.' . $pk . ' IS NULL) AND (c.'
+			. $langField . ' != l.lang_code OR c.' . $langField . ' IS NULL)'
```
Comment (chmst) [inline administrator/components/com_associations/Model/AutoModel.php]: quote l.lang_code and c.extension too

**Comment by rdeutz** [comment]
make sure your PR only includes the needed changes and not branch sync stuff, anything with more than 20 changes files is hard to review

**Comment by Wang-Yu-Chao** [comment]
@rdeutz 
Is making a new PR a good idea?

**Comment by rdeutz** [comment]
@Wang-Yu-Chao very close to a brilliant idea :-)

**Comment by Wang-Yu-Chao** [comment]
@rdeutz Thanks! :)

**Pull request closed without merging**